File: GFG POTD/021-August-2025/012-shop-in-candy-store.java
// https://www.geeksforgeeks.org/problems/shop-in-candy-store1145/1




import java.util.*;

class Solution {
    // TC : O(N*log(N))
    // SC : O(1)
    public ArrayList<Integer> minMaxCandy(int[] prices, int k) {
        int n = prices.length;
        
        Arrays.sort(prices);
        
        int minAmount = 0, maxAmount = 0;
        int i = 0, i_ = n - 1, j_ = 0, j = n - 1;
        
        while (i <= i_ || j_ <= j) {
            if (i <= i_) {
                minAmount += prices[i];
                i_ -= k;
                i ++;
            } 
            if (j_ <= j) {
                maxAmount += prices[j];
                j_ += k;
                j --;
            }
        }
        
        return new ArrayList<>(Arrays.asList(minAmount, maxAmount));
    }
}

File: GFG POTD/021-August-2025/012-shop-in-candy-store.cpp
// https://www.geeksforgeeks.org/problems/shop-in-candy-store1145/1




#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // TC : O(N*log(N))
    // SC : O(1)
    vector<int> minMaxCandy(vector<int>& candies, int K) {
        int N = candies.size();
        
        sort(begin(candies), end(candies));
        
        int mini = 0;
        int buy = 0;
        int free = N-1;
        
        while(buy <= free) {
            mini += candies[buy];
            buy++;
            free -= K;
        }
        
        int maxi = 0;
        buy = N-1;
        free = 0;
        
        while(free <= buy) {
            maxi += candies[buy];
            buy--;
            free += K;
        }
        
        vector<int>ans;
        ans.push_back(mini);
        ans.push_back(maxi);
        return ans;
    }
};